### PR TITLE
Add Mage

### PR DESCRIPTION
**Pull request by Ultradragon005**
idea is by both me & https://discord.com/channels/1094344790910455908/1158325573974757418; cuz I added alot of new stuff not mentioned in the thread

gonna wait for customnetobject merge cuz I wanna make the warp visible ngl

### CODE REVIEW

**Comment by Ultradragon005** [comment]
Maybe I'll make the doubletrigger thing a class everyone can access. Altough I think it's only usefull for like switching stuff and single trigger being the ability, and it also needs a 1.4f delay cuz of minimum 1s shapeshifting delay set by AU.

File: Roles/Crewmate/Mage.cs [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+namespace TOHE.Roles.Crewmate;
+
+internal class Mage : RoleBase
+{
+    //===========================SETUP================================\\
+    private const int Id = 29600;
+    public static bool HasEnabled => CustomRoleManager.HasEnabled(CustomRoles.Mage);
+    public override CustomRoles ThisRoleBase => CustomRoles.Shapeshifter;
+    public override Custom_RoleType ThisRoleType => Custom_RoleType.CrewmatePower;
+    //==================================================================\\
```
Comment (Tommy-XL) [inline Roles/Crewmate/Mage.cs:21]: Wait, where `IsDesyncRole`

**Comment by Tommy-XL** [comment]
Set id 29520 for this role

**Comment by Tommy-XL** [comment]
Wait, I wrote the wrong Id, lool

**Comment by Tommy-XL** [comment]
Use 29700 xd

**Pull request merged**